src/osvr/Util/Logger.cpp

```diff
@@ -64,24 +65,34 @@ namespace util {
         Logger::~Logger() {}
 
         LoggerPtr Logger::makeFallback(std::string const &name) {
-            /// Passes an empty spdlog pointer.
-            std::cerr << "WARNING: logger created for '" << name
-                      << "' is a \"fallback\" logger - an internal error has "
+            // First, we'll attempt to create a console logger to use as a
+            // fallback. If that fails, we'll create a do-nothing logger
+            // instead.
+            std::cerr << "WARNING: Logger created for '" << name
+                      << "' is a 'fallback' logger -- an internal error has "
                          "prevented a standard logger from being created. "
                          "Please report this issue in OSVR-Core on GitHub."
                       << std::endl;
-            return makeLogger(name, std::shared_ptr<spdlog::logger>{});
+            try {
+                auto console_logger = spdlog::stderr_logger_mt(name, false);
+                return makeLogger(name, console_logger);
+            } catch (...) {
+                std::cerr << "Failed to create a console logger to use as a "
+                             "fallback. Logging will be disabled entirely."
+                          << std::endl;
+                auto null_sink = std::make_shared<spdlog::sinks::null_sink_st>();
+                auto null_logger = std::make_shared<spdlog::logger>(name, null_sink);
+                return makeLogger(name, null_logger);
```

Comment:
yeah, I didn't like it either, but I wanted to never fail, and the way we were basically just wrapping spdlog classes meant there needed to be checks all over. Fortunately your changes made it easier to have many fewer tests and still have an absolute fallback.